pkg/controller/operconfig/ingressconfig_controller.go

```diff
@@ -0,0 +1,115 @@
+
+// default ingress controller name
+const DefaultIngressControllerName = "default"
+
+// Reconcile sets the openshift-host-network namespaces' labels as per the
+// endpointPublishingStrategy of the `default` ingress controller object.
+func (r *ReconcileIngressConfigs) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	if request.Namespace == IngressControllerNamespace && request.Name == DefaultIngressControllerName {
+		log.Printf("Reconciling update to %s/%s\n", request.Namespace, request.Name)
+		ingressControllerConfig := &operv1.IngressController{TypeMeta: metav1.TypeMeta{APIVersion: operv1.GroupVersion.String(), Kind: "IngressController"}}
+		err := r.client.Get(context.TODO(), request.NamespacedName, ingressControllerConfig)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				log.Printf("Ingress Controller configuration %s was deleted", request.NamespacedName.String())
+				// Request object not found, could have been deleted after reconcile request.
+				// Owned objects are automatically garbage collected, since we set
+				// the ownerReference (see https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/).
+				// Return and don't requeue
+				return reconcile.Result{}, nil
+			}
+			// Error reading the object - requeue the request.
+			log.Printf("Unable to retrieve IngressController.operator.openshift.io object: %v", err)
+			return reconcile.Result{}, err
+		}
+		labels := map[string]string{HostNetworkLabel: ""}
+		log.Printf("Ingress Controller Config: %+v", ingressControllerConfig)
+		if ingressControllerConfig.Spec.EndpointPublishingStrategy != nil &&
+			ingressControllerConfig.Spec.EndpointPublishingStrategy.Type == operv1.HostNetworkStrategyType {
+			labels[HostNetworkLabel] = HostNetworkLabelValue
+		}
+		err = r.setLabelsOnNamespace(HostNetworkNamespace, labels)
+		if err != nil {
+			log.Printf("Error setting the host network label on namespace %s: %v", HostNetworkNamespace, err)
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{RequeueAfter: ResyncPeriod}, nil
+	}
+	return reconcile.Result{}, nil
+}
+
+// setLabelsOnNamespace sets the labels specified on the target namespace using the client API
+func (r *ReconcileIngressConfigs) setLabelsOnNamespace(targetNamespace string, labels map[string]string) error {
+	var err error
+	namespace := &corev1.Namespace{TypeMeta: metav1.TypeMeta{APIVersion: corev1.SchemeGroupVersion.String(), Kind: "Namespace"}}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: targetNamespace}, namespace)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Printf("Namespace %s was deleted", targetNamespace)
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected, since we set
+			// the ownerReference (see https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/).
+			// Return and don't requeue
```

Comment:
That comment doesn't apply to this code; `targetNamespace` isn't the object we were asked to reconcile, it's an unrelated object that we just expected would exist.

And also, we can't just expect it to always exist anyway, since it will only exist for openshift-sdn and ovn-kubernetes.

I guess since we don't let you change network plugins at runtime, the right fix for that might be that we should only create the ingress config reconciler if the network plugin is sdn or ovn? (It might be nicer if, rather than hardcode in the code the fact that it's those two plugins, we could say "create the ingress config reconciler if the operconfig reconciler reconciles a Namespace whose `.Name == HostNetworkNamespace`, but that implies we're creating the ingress config reconciler after the operconfig reconciler starts running and I'm not sure how easy that would be...)

Reply:
For now, I think it's fine to create the reconciler for all plugin types and just apply the labels if the openshift-host-network namespace exists. We will of course only create the namespace (render the host-network dir) if the plugin type is ovn-k or openshift-sdn.